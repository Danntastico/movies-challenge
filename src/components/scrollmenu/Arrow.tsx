import React, { FunctionComponent } from 'react'

interface ArrowProps {
    text: string;
    classNm: string; //ClassName
}

export const Arrow: FunctionComponent<ArrowProps> = ({text, classNm}) => {
    return (
        <div className={classNm}> 
            {text}
        </div>
    )
}
