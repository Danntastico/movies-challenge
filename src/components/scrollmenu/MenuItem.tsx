import React, { FunctionComponent } from 'react'

interface MenuItemProps {
    text: string;
    selected: string;
}

export const MenuItem: FunctionComponent<MenuItemProps> = ({text, selected}) => {
    return (
        <div className={`menu-item ${selected ? 'active' : ''}`}>
            {text}
        </div>
    )
}
