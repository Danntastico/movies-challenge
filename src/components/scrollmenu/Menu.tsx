import React from 'react'
import { MenuItem } from './MenuItem';

export const Menu = (list: Array<any>, selected:string) =>  (
    list.map( el => {
        const {name} = el;
        return <MenuItem text={name} key={name} selected={selected} />
    })
)