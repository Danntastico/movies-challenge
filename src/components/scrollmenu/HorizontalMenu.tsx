import React, { FunctionComponent, useState } from 'react';

import { useHistory, withRouter } from "react-router-dom";
import ScrollMenu from 'react-horizontal-scrolling-menu'
import { scrollMenuItems } from 'utils/lists'
import { Menu } from './Menu';
import { Arrow } from './Arrow';

const HorizontalMenu: FunctionComponent = () => {
    const history = useHistory();
    const initialItemSelected = { selected: 'TOP 100'};
    const ArrowLeft = Arrow({text: '<', classNm: 'arrow-prev'})
    const ArrowRight = Arrow({text: '>', classNm: 'arrow-next'})

    const menuItems = Menu(scrollMenuItems, initialItemSelected.selected );
    const [select, setSelect] = useState(initialItemSelected)

    const handleOnSelect = (selectedItemKey: string ) => {
        setSelect({selected: selectedItemKey as string});
        const route = selectedItemKey.replace(/\s/g,'') as string;
        history.push(route === 'TOP100'? "/" : `/${route.toLowerCase()}`);
    }

    return (
        <>
            <ScrollMenu 
                data={menuItems}
                arrowLeft={ArrowLeft}
                arrowRight={ArrowRight}
                selected={select.selected}
                onSelect={handleOnSelect as () => void}
            />
        </>
    )
}

export default withRouter(HorizontalMenu)