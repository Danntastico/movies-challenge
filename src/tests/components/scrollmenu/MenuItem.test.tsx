import React from 'react'
import { mount } from 'enzyme';
import { MenuItem } from 'components/scrollmenu/MenuItem'

describe('Unit tests on MenuItem', () => {
    const shall = mount(<MenuItem text="Test text" selected="Item 1" />)

    test('Should render correctly ', () => {
        expect(shall).toMatchSnapshot();
    });
    test('Should render text correctly', () => {
        const menuItemText = shall.find('div').text();

        expect(menuItemText).toEqual('Test text');
    });
});