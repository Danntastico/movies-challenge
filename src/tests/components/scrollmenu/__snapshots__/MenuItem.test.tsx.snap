// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests on MenuItem Should render correctly  1`] = `
<div
  className="menu-item active"
>
  Test text
</div>
`;
